packages/web/src/pages/sign-up-page/pages/CreatePasswordPage.tsx

```diff
@@ -38,47 +64,294 @@ type CreatePasswordValues = {
+        }))
+      }
+    },
+    []
+  )
+
+  const handleConfirmPasswordChange = useCallback(
+    ({ password, confirmPassword }: CreatePasswordValues) => {
+      if (
+        requirementsStatuses.matches !== 'incomplete' ||
+        password.length <= confirmPassword.length
+      ) {
+        setRequirementsStatuses((statuses) => ({
+          ...statuses,
+          matches: getMatchRequirementStatus({ password, confirmPassword })
+        }))
+      }
+    },
+    [requirementsStatuses.matches]
+  )
+
+  const handleConfirmPasswordBlur = useCallback(
+    (values: CreatePasswordValues) => {
+      if (values.password && values.confirmPassword) {
+        setRequirementsStatuses((statuses) => ({
+          ...statuses,
+          matches: getMatchRequirementStatus({
+            password: values.password,
+            confirmPassword: values.confirmPassword,
+            enforceConfirmPasswordNotEmpty: true
+          })
+        }))
+      }
+    },
+    []
+  )
+
+  const hasPasswordError = requirements.some(
+    (r) => r.status === 'error' && r.path === 'password'
+  )
+
+  const hasConfirmPasswordError = requirements.some(
+    (r) => r.status === 'error' && r.path === 'confirmPassword'
+  )
+
+  const isValid = !hasPasswordError && !hasConfirmPasswordError
+
   return (
-    <div>
-      <h1>{messages.header}</h1>
-      <p>{messages.description}</p>
-
-      <p>{messages.yourEmail}</p>
-      <p>{email}</p>
-
-      <Formik initialValues={initialValues} onSubmit={handleSubmit}>
-        <Form>
-          <TextField name='password' label={messages.passwordLabel} />
-          <TextField
-            name='confirmPassword'
-            label={messages.confirmPasswordLabel}
-          />
-          <Button
-            variant={ButtonType.PRIMARY}
-            text={messages.continue}
-            type='submit'
-          />
-        </Form>
-      </Formik>
-    </div>
+    // TODO: Remove all props and styles expect `direction='column'` when this page is plugged into sign up modal.
+    <Flex
+      w={480}
+      h={'calc(100vh - 88px)'}
+      p={'3xl'}
+      className={styles.contentContainer}
+      direction='column'
+    >
+      <Box>
+        <Box mt='xl'>
+          <Text color='heading' size='l' strength='default' variant='heading'>
+            {messages.createYourPassword}
+          </Text>
+        </Box>
+        <Box mt='l'>
+          <Text color='default' size='l' variant='body'>
+            {messages.description}
+          </Text>
+        </Box>
+        <Box mt='2xl'>
+          <Text variant='label' size='xs'>
+            {messages.yourEmail}
+          </Text>
+          <Text variant='body' size='m'>
+            {/* TODO(nkang): Remove `||` once email page completed */}
+            {emailField?.value || 'email@email.com'}
+          </Text>
+        </Box>
+      </Box>
+      <Box mt='l' className={styles.formOuterContainer}>
+        <Formik initialValues={initialValues} onSubmit={handleSubmit}>
```

Comment:
Note that I purposefully opted not to use Zod here or Formik's `validate` prop since we are running validations manually in order to display those completion checkmark states for each requirement.

Reply:
I actually started coding this with Zod but then it started to be more cumbersome than helpful because of the way we show the different password requirements. Basically the Formik error context is not flexible enough to extract the data we need to render the validation elements correctly. Also we need different validations to run on each field's `onChange` and `onBlur`, whereas the Formik validation can only be applied universally. Can explain further if this doesn't make sense!